bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsBridgeHandler.java

```diff
@@ -0,0 +1,880 @@
+
+    private int currentScene;
+
+    // We only keep 2 gson adapters used to serialize and deserialize all messages sent and received
+    private final Gson gsonOut = new Gson();
+    private Gson gsonIn = new Gson();
+
+    public ElroConnectsBridgeHandler(Bridge bridge, NetworkAddressService networkAddressService,
+            ElroConnectsDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(bridge);
+        this.networkAddressService = networkAddressService;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+
+        resetScenes();
+    }
+
+    @Override
+    public void initialize() {
+        ElroConnectsBridgeConfiguration config = getConfigAs(ElroConnectsBridgeConfiguration.class);
+
+        connectorId = config.connectorId;
+        refreshInterval = config.refreshInterval;
+        if (connectorId.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Device ID not set");
+            return;
+        }
+
+        queryString = QUERY_BASE_STRING + connectorId;
+
+        scheduler.submit(this::startCommunication);
+    }
+
+    @Override
+    public void dispose() {
+        stopCommunication();
+    }
+
+    private synchronized void startCommunication() {
+        if (this.connectorId.isEmpty()) {
+            // Should not happen, as this is set in initialize()
+            return;
+        }
+
+        InetAddress addr = null;
+        try {
+            addr = getAddr();
+        } catch (IOException e) {
+            logger.warn("Communication error when finding IP address.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error trying to find IP address for connector ID " + connectorId + ".");
+            stopCommunication();
+            return;
+        }
+        if (addr == null) {
+            logger.warn("No IP address for connector found.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error trying to find IP address for connector ID " + connectorId + ".");
+            stopCommunication();
+            return;
+        }
+
+        String ctrlKey = this.ctrlKey;
+        if (ctrlKey.isEmpty()) {
+            logger.warn("Error initializing communication, no key received.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication data error while starting communication.");
+            stopCommunication();
+            return;
+        }
+
+        DatagramSocket socket;
+        try {
+            socket = createSocket(false);
+            this.socket = socket;
+        } catch (IOException e) {
+            logger.warn("Error initializing communication, could not create socket.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Socket error while starting communication.");
```

Comment:
and here, please go through all uses of the logger and if your changing the status of the thing, remove the logger line.